### ISSUE TITLE
IParameterConsumer does not work with @Option in @ArgGroup

### ISSUE THREAD

**Issue by masupilami** [remkop/picocli#1004]
When using a parameter consumer on an option inside an argument group, the @ArgGroup-annotated field is not instantiated.
Running this test with version 4.2.0
```java
public class Test implements Runnable {

  @ArgGroup
  Group group;

  static class Group {
    @Option(names = "-foo", parameterConsumer = TestConsumer.class)
    String foo;
  }

  static class TestConsumer implements IParameterConsumer {
    @Override
    public void consumeParameters(Stack<String> args, ArgSpec argSpec, CommandSpec commandSpec) {
      argSpec.setValue(args.pop());
    }
  }

  public static void main(String[] args) {
    new CommandLine(new Test()).execute("-foo", "value");
  }

  @Override
  public void run() {}
}
```
throws:
```
PicocliException: Could not set value for field java.lang.String Test$Group.foo to value while processing argument at or before arg[1] 'value' in [-foo, value]: picocli.CommandLine$PicocliException: Could not set value for field java.lang.String Test$Group.foo to value
```
caused by a NPE at line 10145 in picocli.CommandLine.Model.FieldBinding.set(T).
Is this the intended behavior or a bug?
thanks

**Comment by remkop** [comment]
Thank you for raising this! 
I will investigate when I get to my PC.

**Comment by remkop** [comment]
If you need a quick workaround, I suspect you can avoid the NPE if you instantiate the group at the field declaration:

```java
  @ArgGroup
  Group group = new Group();
```

I need to investigate further later.

**Comment by remkop** [comment]
Fixed in master.
The fix will be part of the upcoming picocli 4.3 release.

Thank you again for raising this!
